fw/vhost.c

```diff
@@ -2056,6 +2069,52 @@ tfw_cfgop_frang_glob_set_int(TfwCfgSpec *cs, TfwCfgEntry *ce)
 	return tfw_cfg_set_int(cs, ce);
 }
 
+static int
+tfw_cfgop_frang_glob_set_long(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	if (ce->dflt_value && *(unsigned long *)(cs->dest))
+		return 0;
+	return tfw_cfg_set_long(cs, ce);
+}
+
+static int
+tfw_cfgop_frang_glob_http_methods(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	long int *dest_long = cs->dest;
+
+	if (ce->dflt_value && *(unsigned long *)(cs->dest))
+		return 0;
+	return __tfw_cfgop_frang_http_methods(cs, ce, dest_long);
+}
+
+static int
+tfw_cfgop_frang_glob_http_ct_vals(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	FrangVhostCfg *cfg = &tfw_frang_vhost_reconfig;
+
+	if (cfg->http_ct_vals) {
+		if (ce->dflt_value)
+			return 0;
+		kfree(cfg->http_ct_vals);
+		cfg->http_ct_vals = NULL;
+	}
+	return __tfw_cfgop_frang_http_ct_vals(cs, ce, cfg);
+}
+
+static int
+tfw_cfgop_frang_glob_rsp_code_block(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	FrangVhostCfg *cfg = &tfw_frang_vhost_reconfig;
+
+	if (cfg->http_resp_code_block) {
+		if (ce->dflt_value)
+			return 0;
+		kfree(cfg->http_resp_code_block);
+		cfg->http_resp_code_block = NULL;
+	}
+	return __tfw_cfgop_frang_rsp_code_block(cs, ce, cfg);
```

Comment:
Can you also check `__tfw_cfgop_frang_rsp_code_block`. I see that we return -EINVAL after memory allocation without
freeing previously allocated cb = kzalloc(sizeof(FrangHttpRespCodeBlock), GFP_KERNEL); Is there memory leak or we will free it later?

Reply:
It assigns to conf->http_resp_code_block and will be freed by __tfw_frang_clean() in case of error.